protocol/streaming/grpc/grpc_streaming_manager.go

```diff
@@ -22,8 +23,14 @@ type GrpcStreamingManagerImpl struct {
 
 // OrderbookSubscription represents a active subscription to the orderbook updates stream.
 type OrderbookSubscription struct {
+	// Indicate whether the subscription has been initialized with a snapshot.
+	initialized bool
```

Comment:
If this works, it works, but I think `sync.Once` is a cleaner/more performant way to do this singleton/initialize pattern. If we do that we don't have to keep locking and unlocking on every subscription in `GetUninitializedClobPairIds()`.  We can just call the sync.once subscription for every subscription in the orderbook subscription array and be sure the code only runs once.

Maybe timebox exploring this approach? but i don't feel too strongly if this approach works since this is only run on full nodes (which I assume don't have to be as performant as validators) so feel free to ignore.